Jenkinsfile

```diff
@@ -926,9 +926,39 @@ pipeline {
                             unitTestPost()
                         }
                     }
-                }
-            }
-        }
+                } // End run_test.sh stage
+                stage('run_test.sh with memcheck') {
```

Comment:
Stage name needs to be "Unit Test with memcheck"  and this is what parseStageInfo in pipeline-lib should be using to parse the defaults.

This PR should not be landed before https://github.com/daos-stack/daos/pull/3104, please use that in-progress PR as a guide.

It has changes that are needed for running multiple valgrind stages in parallel, with the valgrind publishing as the last step of the Jenkinsfile.

Reply:
> It has changes that are needed for running multiple valgrind stages in parallel, with the valgrind publishing as the last step of the Jenkinsfile.

That seems an odd requirement, what particular problem is it trying to solve?

Reply:
To establish a patter to future proof our code.
It is easier for us to add VMs to CI than physical nodes.
And running VM tests in parallel where we can is a way to speed up the CI turnaround.

The Unit Test stage must complete before the functional tests can start.

The process of adding the Bullseye tests showed this issue.
The existing unit tests are doing some valgrind tests.
Bullseye stage is just running everything the unit tests with bullseye enabled.
This initially resulted with two attempts to publish valgrind reports.
Currently in the bullseye PR, the extra valgrind reports are just ignored.